pystac/asset.py

```diff
@@ -0,0 +1,165 @@
+from copy import copy
+from typing import Any, Dict, List, Optional, TYPE_CHECKING, Union
+
+import pystac as ps
+from pystac.utils import is_absolute_href, make_absolute_href
+
+if TYPE_CHECKING:
+    from pystac.collection import Collection as Collection_Type
+    from pystac.item import Item as Item_Type
+
+
+class Asset:
+    """An object that contains a link to data associated with an Item or Collection that
+    can be downloaded or streamed.
+
+    Args:
+        href (str): Link to the asset object. Relative and absolute links are both
+            allowed.
+        title (str): Optional displayed title for clients and users.
+        description (str): A description of the Asset providing additional details,
+            such as how it was processed or created. CommonMark 0.29 syntax MAY be used
+            for rich text representation.
+        media_type (str): Optional description of the media type. Registered Media Types
+            are preferred. See :class:`~pystac.MediaType` for common media types.
+        roles ([str]): Optional, Semantic roles (i.e. thumbnail, overview,
+            data, metadata) of the asset.
+        properties (dict): Optional, additional properties for this asset. This is used
+            by extensions as a way to serialize and deserialize properties on asset
+            object JSON.
+
+    Attributes:
+        href (str): Link to the asset object. Relative and absolute links are both
+            allowed.
+        title (str): Optional displayed title for clients and users.
+        description (str): A description of the Asset providing additional details,
+            such as how it was processed or created. CommonMark 0.29 syntax MAY be
+            used for rich text representation.
+        media_type (str): Optional description of the media type. Registered Media Types
+            are preferred. See :class:`~pystac.MediaType` for common media types.
+        properties (dict): Optional, additional properties for this asset. This is used
+            by extensions as a way to serialize and deserialize properties on asset
+            object JSON.
+        owner: The Item or Collection this asset belongs to, or None if it has no owner.
```

Comment:
If keeping the type in the doc string is the way to go, this one is missing the type.

Reply:
Hey @schwehr - it seems like your going into some detailed line edits in your review. A lot of these comments are around work that can be managed in a follow-up PR as I mentioned, or are around stylistic and/or relatively non-important coding choices that, while aren't necessarily set for the project, are not intended to be changed in this PR. I appreciate the contribution, but I'd recommend sticking to a higher level review, helping out the broader goal of making sure these changes set us up for a STAC 1.0 release, and putting some follow up issues with some of these finer details, e.g. should we change variable names or change all `if any(x)` checks to `if x`. Thanks!

Reply:
Sorry, I have never done reviews the way you seem to be aiming for.  I am uncomfortable with this giant or going in without careful review first.  I find it leads to long shakeout periods with lingering bugs.  I can't see the big picture through all the little things.  I am used to huge PRs like this that are used as the general target, but then have clean pieces broken out to review. I don't think I can be much help on the level you are looking for.   Unless the process changes, I need to bow out of additional reviews.

Reply:
Thanks for the context. Apologies if the ask for a review was unclear. I can totally see the value of the type of review your doing, and perhaps at another point we should decide on some conventions and modify the codebase to be consistent around them; for this work we're trying to make some major changes that unblock a lot of things moving to STAC 1.0 which we want to happen before the final release, so moving a bit fast. Makes sense that you'd bow out of this one, and hopefully we can get to a better process that's more clear and making use of yours and other's valuable time and effort moving forward.